django/contrib/admin/static/admin/js/change_form.js

```diff
@@ -2,8 +2,21 @@
 {
     const inputTags = ['BUTTON', 'INPUT', 'SELECT', 'TEXTAREA'];
     const modelName = document.getElementById('django-admin-form-add-constants').dataset.modelName;
+    let submitted = false;
+
     if (modelName) {
         const form = document.getElementById(modelName + '_form');
+
+        form.addEventListener('submit', function(event) {
+            event.preventDefault();
+            if (submitted) {
+                const answer = window.confirm('You already submitted. Did you mean to submit again?');
```

Comment:
This will need to be translated – I believe you’ll need `/*global gettext*/` at the top of the file, and here:

```suggestion
                const answer = window.confirm(gettext('You already submitted. Did you mean to submit again?'));
```

I have no experience with this myself, just trying to match a similar implementation at https://github.com/django/django/blob/main/django/contrib/admin/static/admin/js/actions.js#L161.

Reply:
Thank you!!